src/NzbDrone.Core/Download/Aggregation/Aggregators/AggregateLanguages.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NLog;
+using NzbDrone.Core.Languages;
+using NzbDrone.Core.Parser;
+using NzbDrone.Core.Parser.Model;
+
+namespace NzbDrone.Core.Download.Aggregation.Aggregators
+{
+    public class AggregateLanguages : IAggregateRemoteEpisode
+    {
+        private readonly Logger _logger;
+
+        public AggregateLanguages(Logger logger)
+        {
+            _logger = logger;
+        }
+
+        public RemoteEpisode Aggregate(RemoteEpisode remoteEpisode)
+        {
+            var parsedEpisodeInfo = remoteEpisode.ParsedEpisodeInfo;
+            var languages = parsedEpisodeInfo.Languages;
+            var series = remoteEpisode.Series;
+            var releaseTokens = parsedEpisodeInfo.ReleaseTokens ?? parsedEpisodeInfo.ReleaseTitle;
+            var normalizedReleaseTokens = Parser.Parser.NormalizeEpisodeTitle(releaseTokens);
+            var cleanNormalizedReleaseTokens = normalizedReleaseTokens.CleanSeriesTitle();
+
+            if (series == null)
+            {
+                _logger.Debug("Unable to aggregate languages, using parsed values: {0}", string.Join(", ", languages.ToList()));
+
+                remoteEpisode.Languages = languages;
+
+                return remoteEpisode;
+            }
+
+            // Exclude any languages that are part of the episode title, if the episode title is in the release tokens (falls back to release title)
+            foreach (var episode in remoteEpisode.Episodes)
+            {
+                var episodeTitleLanguage = LanguageParser.ParseLanguages(episode.Title);
+
+                if (!episodeTitleLanguage.Contains(Language.Unknown))
+                {
+                    var normalizedEpisodeTitle = Parser.Parser.NormalizeEpisodeTitle(episode.Title);
+                    var cleanNormalizedEpisodeTitle = normalizedEpisodeTitle.CleanSeriesTitle();
+
+                    if (normalizedReleaseTokens.Contains(normalizedEpisodeTitle, StringComparison.CurrentCultureIgnoreCase) ||
+                        cleanNormalizedReleaseTokens.Contains(cleanNormalizedEpisodeTitle, StringComparison.CurrentCultureIgnoreCase))
+                    {
+                        languages = languages.Except(episodeTitleLanguage).ToList();
```

Comment:
What if the release actually contains the language from the title as well. Should we replace the title then do a reparse to get the new language set instead of just taking it out here.

Reply:
One concern was about accurately removing them, but I think my added commit is close enough for most cases. The only thing that we might want to do to take it further would be to use regex and allow for fuzzier matching, but didn't want to go that route just yet.